### PR TITLE
API-13639 Updated build.sh to run agent j generate xlsx job on lab builds.

### PR DESCRIPTION
**Pull request by taylorrassmannlibertyits**

### CODE REVIEW

**Comment by taylorrassmannlibertyits** [comment]
> Is this on hold until the corresponding agent-j job is ready?
The current master agent-j will simply exit 0 right away so no current xlsx generation will occur if we have a lab build right now.

**Comment by taylorrassmannlibertyits** [comment]
So to answer your question, maybe this is on hold depending on how we want to handle?

**Comment by jblefkove-va** [comment]
> So to answer your question, maybe this is on hold depending on how we want to handle?

I see that the stub of the agent-j job is scheduled for each morning (pending that PR to remove the scheduling), so they can merge in either order

File: build.sh [hunk 1]
```diff
@@ -153,4 +182,10 @@ if [ -n "${RESOURCES:-}" ]; then MVN_ARGS+=" -Dresources=$RESOURCES"; fi
 
 mvn ${MVN_ARGS:-} -Dimport.directory=$DATAMART_DIR -Dconfig.file=$CONFIG_FILE -Dtest=PopulateDb test
 
+if [[ "${ENVIRONMENT}" == "lab" ]]
```
Comment (jblefkove-va) [inline build.sh:185]: We shouldn't check the environment both here and inside the function

File: build.sh [hunk 2]
```diff
@@ -112,6 +112,35 @@ runDataQueryTests() {
   return 0
 }
 
+runGenerateSyntheticHealthcareSpreadsheet() {
+  if [[ "${ENVIRONMENT}" != "lab" ]]
+  then
+    echo "Not running xlsx generation for: $ENVIRONMENT" && return 1
```
Comment (jblefkove-va) [inline build.sh:118]: Let's print a message only when we are starting the job; this one is less helpful, particularly when running locally

File: build.sh [hunk 3]
```diff
@@ -112,6 +112,35 @@ runDataQueryTests() {
...
+  fi
+
+  if [ "$RUN_DATA_QUERY_TESTS" == "true" ]
```
Comment (jblefkove-va) [inline build.sh:121]: remove this check

**Review by jblefkove-va — APPROVED** [review]
:shipit:

**Pull request merged**